components/modbus_controller.rst

```diff
@@ -166,135 +161,262 @@ Some devices use decimal values in read registers to show multiple binary states
+          ESP_LOGD("Modbus Sensor Lambda", "FP32 = 0x%08X => %f", raw_to_float.raw, raw_to_float.float_value);
+          return raw_to_float.float_value;
+        unit_of_measurement: kVArh
+        accuracy_decimals: 1
+
+.. _modbus_register_count:
+
+Optimizing modbus communications
+--------------------------------
+
+``register_count`` is an option only required for uncommon response encodings or to optimizie modbus communications.
+
+It describes the number of registers this data point spans. Overrides the defaults determined by ``value_type``. If no value for ``register_count`` is provided, it is calculated based on the register type. The default size for 1 register is 16 bits (1 Word). Some devices are not adhering to this convention and have registers larger than 16 bits. In this case ``register_count`` and  ``response_size`` must be set. For example, if your modbus device uses 1 registers for a FP32 value instead the default of two set ``register_count: 1`` and ``response_size: 4``.
+
+``register_count`` can also be used to skip a number of registers in consecutive range.
+
+An example is an SDM meter, with interesting data in register addresses 0, 2, 4 and 6:
+
+.. code-block:: yaml
+
+    - platform: modbus_controller
+      name: "Voltage Phase 1"
+      address: 0
+      register_type: "read"
+      value_type: FP32
+
+    - platform: modbus_controller
+      name: "Voltage Phase 2"
+      address: 2
+      register_type: "read"
+      value_type: FP32
+
+    - platform: modbus_controller
+      name: "Voltage Phase 3"
+      address: 4
+      register_type: "read"
+      value_type: FP32
+
+    - platform: modbus_controller
+      name: "Current Phase 1"
+      address: 6
+      register_type: "read"
+      value_type: FP32
+      accuracy_decimals: 1
+
+The configuration above will generate *one* modbus command *read multiple registers from 0 to 6*.
+
+Maybe you don’t care about the data in register addresses 2 and 4, which are voltage values for Phase 2 and Phase 3 (or you have a SDM-120).
+Of course, you can delete the sensors your don’t care about. But then you have a gap in the addresses. If you remove the registers at address 2 and 4 then *two* commands will be generated *read register 0* and *read register 6*. To avoid the generation of multiple commands and reduce the amount of uart communication ``register_count`` can be used to fill the gaps:
+
+.. code-block:: yaml
+
+    - platform: modbus_controller
+      name: "Voltage Phase 1"
+      address: 0
+      unit_of_measurement: "V"
+      register_type: "read"
+      value_type: FP32
+      register_count: 6
+
+    - platform: modbus_controller
+      name: "Current Phase 1"
+      address: 6
+      register_type: "read"
+      value_type: FP32
+
+Because the option `register_count: 6` is used for the first sensor, *one* command *read multiple registers from 0 to 6* will be used but the values in between will be ignored.
```

Comment:
```suggestion
Because the option ``register_count: 6`` is used for the first sensor, *one* command *read multiple registers from 0 to 6* will be used but the values in between will be ignored.
```